transcendence-app/src/shared/types/custom.d.ts

```diff
@@ -3,7 +3,7 @@ import { User } from '../../user/user.domain';
 
 declare module 'http' {
   interface IncomingMessage {
-    user: User | null;
+    user: User;
```

Comment:
The user can't be `null` because the [passport](https://www.passportjs.org/concepts/authentication/downloads/html/#middleware) set it to the user object and we have the [WsAuthenticatedGuard](https://github.com/42AGV/ft_transcendence/blob/4092458a15338b76fafa1eec0a60ddc4cad6f7e7/transcendence-app/src/shared/guards/ws-authenticated.guard.ts#L8) which checks if the user is authenticated or throws an exception if not.